.circleci/config.yml

```diff
@@ -1,241 +1,216 @@
-  filter_only_release_tags: &filter_only_release_tags
-    branches:
-      ignore: /.*/
-    tags:
-      # regex explanation:
-      # "^" asserts position at start of a line.
-      # "v" matches the character "v" literally (case sensitive).
-      # "[0-9]+" match a single numerical digit (the + quantifier, matches between one and unlimited times, as many times as possible).
-      # "-" matches the character "-" literally.
-      # "rc" matches the character "rc" literally (case sensitive).
-      # "$" asserts position at the end of a line.
-      # Workbench release tag format example: v5-3-rc1
-      only: /^v[0-9]+-[0-9]+-rc[0-9]+$/
-
-  # Job runs for PR branches only
-  filter_only_pr_branch: &filter_only_pr_branch
-    tags:
-      ignore: /.*/
-    branches:
-      ignore: master
+  puppeteer-executor:
+    docker:
+      - image: << pipeline.parameters.workbench-image >>
+    working_directory: ~/workbench
+    environment:
+      <<: *puppeteer-environment
 
 # -------------------------
 #   COMMANDS
 # Refers to https://circleci.com/docs/2.0/reusing-config/#authoring-reusable-commands
 # -------------------------
 commands:
-  activate_service_account_credential:
+  gcloud-auth-login:
     description: "Activiate CircleCI service account credential"
     steps:
       - run:
-          name: Activate CircleCI service account credential
           working_directory: ~/workbench
-          # Used to call gsutil from the circle environment.
           command: ci/activate_creds.sh api/circle-sa-key.json
 
-  checkout_init_git:
+  checkout-code:
     description: "git checkout and update submodules"
     steps:
       - checkout
       - run:
-          name: Update git submodules
           command: git submodule update --init --recursive
 
-  halt_if_code_unchanged:
-    description: "Halt job if no code changed"
+  store-commit-message:
+    description: "Export Git commit message to environment variable"
+    steps:
+      - run:
+          command: |
+            echo 'export COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")' >> $BASH_ENV
+            source $BASH_ENV
+          name: "Export COMMIT_MESSAGE environment variable"
+
+  halt-puppeteer-check:
+    description: "Halt if changed file name match ignore patterns"
     steps:
       - run:
+          name: "Halt if changed file name match ignore patterns"
           command: |
-            if ! git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -qvFf .circleci/e2e-job-ignore-patterns.txt; then
-              echo "Application code have not changed. halting job."
-              circleci step halt
-            else
-              git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -vFf .circleci/e2e-job-ignore-patterns.txt
-              echo "job continuing."
+            if ! git diff --name-only $(git merge-base origin/master ${CIRCLE_BRANCH}) | grep -qvFf .circleci/e2e-job-ignore-patterns.txt ; then
```

Comment:
If you need any more complexity here, I'll write you a short Ruby script like the Bash weenie that I am.